slick/src/main/scala/slick/jdbc/JdbcStatementBuilderComponent.scala

```diff
@@ -608,7 +608,7 @@ trait JdbcStatementBuilderComponent { self: JdbcProfile =>
     }
 
     protected def createPhase1 = Iterable(createTable(false)) ++ primaryKeys.map(createPrimaryKey) ++ indexes.map(createIndex)
-    protected def createIfNotExistsPhase = Iterable(createTable(true)) ++ primaryKeys.map(createPrimaryKey) ++ indexes.map(createIndex)
+    protected def createIfNotExistsPhase = Iterable(createTable(true)) ++ primaryKeys.map(createPrimaryKey) ++ indexes.map(createIndex) ++ foreignKeys.map(dropForeignKeyIfExists) ++ foreignKeys.map(createForeignKey)
```

Comment:
That's a good question. I didn't notice but testing the current 3.3.3 version and it also fails if there is an index.
Also, for some reason, primary keys don't get added through that statement. Seems like the `primaryKeys` method in the AbstractTable isn't returning anything. That's why it doesn't fail. It only works because the create table method already includes the primary key definition.

Reply:
To me `drop fk if exists` + `create fk` != `create fk if not exists`. If a foreign key is already present, then it shouldn't be affected by `create (everything) if not exists`.

Reply:
The problem is that there is no way to modify the existing foreign key without recreating it. There's also no `create if not exists` for foreign keys in SQL.

Reply:
Yep, that seems to be correct. I've googled modifying foreign keys, but haven't found anything portable and not requiring special privileges. I've found some solutions specific to particular databases, but that's probably overkill.